packages/components/src/post-locked-modal/index.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { Modal } from '@wordpress/components';
+import { Component, Fragment } from '@wordpress/element';
+import { select } from '@wordpress/data';
+import {
+	PostPreviewButton,
+} from '@wordpress/editor';
+import './style.scss';
+import { addQueryArgs } from '@wordpress/url';
+import { getPostEditURL } from '../../../../edit-post/components/browser-url';
+import { getWPAdminURL } from '../../../editor/src/utils/url';
+
+class PostLockedModal extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			isOpen: true,
+		};
+
+		this.takeOverPost = this.takeOverPost.bind( this );
+		const { getLockDetails, getPostLockUser } = select( 'core/editor' );
```

Comment:
Any reason to avoid using `withSelect` here to call the selectors, this would ensure the component is refreshed oncee the values change. Is this something we don't want to do?

Reply:
nope, i just forgot about withSelect :) I'll rework that.

Reply:
switched to withSelect